### PR TITLE
color: added --c-hom option

### PR DESCRIPTION
**Pull request by tarak77**

### CODE REVIEW

File: color.py [hunk 1]
```diff
@@ -115,14 +115,15 @@ def color(argv):
...
-        opts, args = getopt.getopt(argv[1:], "c:n:l:m:L:i:s:S:hd:r:I:p:P:CD:R", ["min-num=", "missing=", "max-r=", "bin-size="])
+        opts, args = getopt.getopt(argv[1:], "c:n:l:m:L:i:s:S:hd:r:I:p:P:CD:R", ["min-num=", "missing=", "max-r=", "bin-size=", "c-hom="])
     except getopt.GetoptError as err:
         sys.stderr.write("[E::" + __name__ + "] unknown command\n")
         return 1
     if len(args) == 0:
         sys.stderr.write("Usage: dip-c color [options] <in.3dg>\n")
         sys.stderr.write("Options:\n")
         sys.stderr.write("  -c <color.txt>    color by a list of locus-color pairs (tab-delimited: chr, locus, color)\n")
+        sys.stderr.write("  --c-hom=<color.txt>    color by a list of homolog specific locus-color pairs (tab-delimited: chr, locus, color)\n")
```
Comment (tanlongzhi) [inline color.py:126]: Please change "chr, locus, color" to "homolog, locus, color"

File: color.py [hunk 2]
```diff
@@ -199,6 +205,13 @@ def color(argv):
             ref_locus = int(ref_locus)
             color = float(color)
             ref_name_ref_locus_colors[(ref_name, ref_locus)] = color
+    elif color_mode == "c-hom":
+        ref_name_ref_locus_colors = {}
```
Comment (tanlongzhi) [inline color.py:209]: Please change "ref_name_ref_locus_colors" to "hom_name_ref_locus_colors"

File: color.py [hunk 3]
```diff
@@ -199,6 +205,13 @@ def color(argv):
...
+        for color_file_line in color_file:
+            ref_name, ref_locus, color = color_file_line.strip().split("\t")
```
Comment (tanlongzhi) [inline color.py:211]: Please change "ref_name" to "hom_name"

File: color.py [hunk 4]
```diff
@@ -199,6 +205,13 @@ def color(argv):
...
+            ref_locus = int(ref_locus)
+            color = float(color)
+            ref_name_ref_locus_colors[(ref_name, ref_locus)] = color
```
Comment (tanlongzhi) [inline color.py:214]: Please change accordingly

File: color.py [hunk 5]
```diff
@@ -248,6 +261,11 @@ def color(argv):
                 color = ref_name_ref_locus_colors[(g3d_particle.get_ref_name(), g3d_particle.get_ref_locus())]
             except KeyError:
                 continue
+        elif color_mode == "c-hom":
+            try:
+                color = ref_name_ref_locus_colors[(g3d_particle.get_hom_name(), g3d_particle.get_ref_locus())]
```
Comment (tanlongzhi) [inline color.py:266]: Please change accordingly

**Review by tanlongzhi — CHANGES_REQUESTED** [review]
Please see my inline comments to change variable names for clarity

**Comment by tarak77** [comment]
made changes

**Review by tanlongzhi — APPROVED** [review]

**Pull request merged**